File: haohaninfo/交易實務案例/110.R
#取得即時報價，詳細在技巧51
source("function.R")
source("order.R")

#設定初始倉位，若為0則為無在倉部位
index <- 0

#設定停損停利點
stopLoss <- 10
takeProfit <- 10

#設定進出場時機點
orderPrice <- 0
coverPrice <- 0
orderTime <- strptime('08:50:00.00','%H:%M:%OS')
coverTime <- strptime('09:00:00.00','%H:%M:%OS')

#進場條件判斷
while (index==0){
 #取得及時成交資訊
 Mdata<-GetMatchData(DataPath,Date)
 #因為要進行比較，所以透過R內建函數strptime轉為R語言的時間格式
 #最小單位可至秒數以下兩位
 MatchTime <- strptime(Mdata[[1]][1],'%H:%M:%OS')
 MatchPrice <- as.numeric(Mdata[[1]][2])

 if(MatchTime >= orderTime){
  orderInfo <- OrderMKT('TX00','B',1)
  orderPrice <- as.numeric(strsplit(orderInfo,",")[[1]][5])
  index <- 1
  print(paste("Order Buy Success Price:",orderPrice))
  break
 }
 
}

#出場條件判斷
while (index!=0){
 #取得及時成交資訊
 Mdata<-GetMatchData(DataPath,Date)
 #因為要進行比較，所以透過R內建函數strptime轉為R語言的時間格式
 #最小單位可至秒數以下兩位
 MatchTime <- strptime(Mdata[[1]][1],'%H:%M:%OS')
 MatchPrice <- as.numeric(Mdata[[1]][2])

 if(MatchPrice > orderPrice + takeProfit | MatchPrice + stopLoss < orderPrice | MatchTime >= coverTime ){
  coverInfo <- OrderMKT('TX00','S',1)
  coverPrice <- as.numeric(strsplit(coverInfo,",")[[1]][5])
  index <- 0
  print(paste("Cover Buy Success Price:",coverPrice,"Profit:",coverPrice-orderPrice))
  break
 }
}

File: haohaninfo/即時範例/110.py
# -*- coding: UTF-8 -*-

#取得報價資訊，詳情請查看技巧51
execfile('function.py')
#取得下單函數，詳情請查看技巧103
execfile('order.py')

#設定開始時間及結束時間
startTime=datetime.datetime.strptime('09:00:00.00',"%H:%M:%S.%f")
endTime=datetime.datetime.strptime('10:00:00.00',"%H:%M:%S.%f")

#設定初始倉位，若為0則為無在倉部位
index=0
orderTime=0
orderPrice=0
coverTime=0
coverPrice=0
#定義指標變數
stopLoss=10
takeProfit=10


#進場判斷
for i in getMatch():		
 MatchInfo=i.split(',')
 MatchTime=datetime.datetime.strptime(MatchInfo[0],"%H:%M:%S.%f")
 MatchPrice=int(MatchInfo[1])
 #時間到則進場
 if MatchTime>=startTime:
  index=1
  orderInfo=OrderMKT('TX00','B','1')
  orderTime=orderInfo[6]
  orderPrice=int(orderInfo[4])
  print orderTime,"Order Buy Success! Price:",orderPrice
  break

#出場判斷 
for i in getMatch():		
 MatchInfo=i.split(',')
 MatchTime=datetime.datetime.strptime(MatchInfo[0],"%H:%M:%S.%f")
 MatchPrice=int(MatchInfo[1])
 
 if index==1:
  #停損停利判斷
  if MatchPrice>=orderPrice+takeProfit or MatchPrice<=orderPrice-stopLoss:
   index=0
   coverInfo=OrderMKT('TX00','S','1')
   coverTime=coverInfo[6]
   coverPrice=int(coverInfo[4])
   print coverTime,"Order Sell Success! Price:",coverPrice
   break
  #時間到則出場
  if MatchTime>=endTime:
   index=0
   coverInfo=OrderMKT('TX00','S','1')
   coverTime=coverInfo[6]
   coverPrice=int(coverInfo[4])
   print coverTime,"Order Sell Success! Price:",coverPrice
   break 

